interface/modules/zend_modules/module/Installer/src/Installer/Controller/InstallerController.php

```diff
@@ -140,23 +143,122 @@ public function manageAction()
+                        $prev_html_tag = false;
+                        if (preg_match("/<([a-z]+).*?>([^<]+)<\/([a-z]+)>/i", $string, $mm)) {
+                            if ($add_query_string > 0) {
+                                $div[] = "</div>";
+                                $add_ended_divs++;
+                            }
+                            $div[] = $string;
+                            $prev_html_tag = true;
+                            $curr_html_tag = true;
+                        }
+                        if (!$prev_html_tag && $curr_html_tag) {
+                            $div[] = "<div class='show_hide_log'>".xlt("show/hide executed query log")."</div><div class='spoiler' style='margin-left: 10px' >".$string;
+                            $curr_html_tag = false;
+                        } elseif (!$prev_html_tag && !$curr_html_tag) {
+                            $div[] = $string;
+                            $add_query_string++;
+                        }
+                        if (count($matches[1]) == (count($div) - $add_ended_divs) && (!$prev_html_tag && !$curr_html_tag )) {
+                            $div[] = "</div>";
+                        }
+                        $k++;
+                    }
+                }
+                $this -> getInstallerTable() -> updateRegistered($request->getPost('modId'), '', $values);
+
+                $status = $this->listenerObject->z_xlt("Success");
+            } elseif ($request->getPost('modAction') == 'install_acl') {
+                $Module = $this->getInstallerTable()->getRegistryEntry($request->getPost('modId'), "mod_directory");
+                $modDir = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir']."zend_modules/module/".$Module -> modDirectory;
+                if (file_exists($modDir."/acl/acl_setup.php") && empty($modDir->acl_version)) {
+                    ob_start();
+                    include_once($modDir."/acl/acl_setup.php");
+                    $div[] = ob_get_contents();
+                    ob_end_clean();
+                    $values[2] = $Module->sql_version;
+                    $values[3] = $this->getModuleVersionFromFile($request->getPost('modId'));
+                    $this -> getInstallerTable() -> updateRegistered($request->getPost('modId'), '', $values);
+                    $status = $this->listenerObject->z_xlt("Success");
+                }
+            } elseif ($request->getPost('modAction') == 'upgrade_acl') {
+                $Module = $this->getInstallerTable()->getRegistryEntry($request->getPost('modId'), "mod_directory");
+                $modDir = $GLOBALS['srcdir']."/../".$GLOBALS['baseModDir']."zend_modules/module/".$Module -> modDirectory;
+                if (file_exists($modDir."/acl/acl_upgrade.php") && !empty($Module->acl_version)) {
+                    global $ACL_UPGRADE;
+                    ob_start();
+                    include_once($modDir."/acl/acl_upgrade.php");
```

Comment:
when directly calling these, we should incorporate a mechanism to prevent folks from trying to call these scripts directly (since we can't feasibly remove them all like we recommend for the main acl_upgrade/acl_setup scripts). See what was done in the Administraion->System->Language to prevent this:
https://github.com/openemr/openemr/blob/master/interface/language/language.php#L53-L56
https://github.com/openemr/openemr/blob/master/interface/language/lang_language.php#L15-L21

Reply:
Fixed